openhtf/util/validators.py

```diff
@@ -235,7 +235,7 @@ class WithinPercent(RangeValidatorBase):
   def __init__(self, expected, percent):
     if percent < 0:
       raise ValueError('percent argument is {}, must be >0'.format(percent))
-    self.expected = expected
+    self.expected = abs(expected)
```

Comment:
Don't use the absolute value of the expected like this.  It would accept the wrongly signed value.

Instead, just absolute value to the applied percent and then add/subtract that as indicated.

`@property`
`def _applied_percent(self):`
`  return abs(self.expected * self.percent / 100.0)`
` `
`@property`
`def minimum(self):`
`  return self.expected - self._applied_percent`
` `
`@property`
`def maximum(self):`
`  return self.expected + self._applied_percent`

Reply:
ahh, you're right.